pkg/cmd/cluster/cluster_attach.go

```diff
@@ -13,11 +18,62 @@ func newCmdClusterAttach(ctx api.Context) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			manager := ctx.ConfigManager()
 			conf, err := manager.Find(args[0], false)
+
+			// If we have not found any config, we will look in the linked clusters.
+			if err != nil && err != config.ErrConfigNotFound {
+				return err
+			}
+
+			if conf != nil {
+				return manager.Attach(conf)
```

Comment:
If the search matches both a locally configured cluster and a linked cluster, this would attach to the local one. I'd assume this should instead fail with an error saying there are multiple matches.